src/js/server/middleware/applications-middleware.ts

```diff
@@ -4,28 +4,71 @@ import * as statuses from 'js/shared/status-constants';
 /**
  * Intercepts the request to check if the user has submitted an application
  * 
- * If they have, it will redirect them to the dashboard. Otherwise, it will let them proceed
+ * If they have, it will redirect them to the previous page. Otherwise, it will let them proceed
  * as normal.
  */
-export function goHomeIfAlreadyApplied(req, res, next) {
-  req.user.getHackerApplication().then((hackerApplication: HackerApplicationInstance) => {
-    if (hackerApplication) {
-      res.redirect(`${req.baseUrl}/dashboard`);
-      return;
-    }
-    next();
+ 
+export function goBackIfApplied(req, res, next) {
```

Comment:
Please could you add type annotations for the arguments and return type of new functions, this will help gradually improve the safety of our codebase.  Some of these functions look like they should return booleans but seem to return `boolean | void`, adding type annotations would catch this.